application/language/german/kalkun_lang.php

```diff
@@ -3,358 +3,372 @@
+$lang['Outbox'] = 'Ausgang';
+$lang['Sent items'] = 'Gesendet';
+$lang['Sending error'] = 'Sending Error';
+$lang['Trash'] = 'Papierkorb';
+$lang['Spam'] = 'Spam';
+$lang['Add'] = 'Neu';
+$lang['Action'] = 'Aktion';
+
+$lang['Phonebook'] = 'Adressbuch';
+$lang['Settings'] = 'Einstellungen';
+$lang['Filters'] = 'Filters';
+$lang['About'] = 'About';
+$lang['Logout'] = 'Abmelden';
 
 // Dashboard
-$lang['kalkun_statistic'] = 'Statistik';
-$lang['kalkun_system_information'] = 'System Information';
-$lang['kalkun_alerts'] = 'Alarm';
-$lang['kalkun_operating_system'] = 'Betriebssystem';
-$lang['kalkun_gammu_version'] = 'Gammu Version';
-$lang['kalkun_gammu_db_schema'] = 'Gammu DB Schema';
-$lang['kalkun_phone_imei'] = 'Modem IMEI';
+$lang['Statistics'] = 'Statistik';
+$lang['System information'] = 'System Information';
+$lang['Alerts'] = 'Alarm';
+$lang['Operating system'] = 'Betriebssystem';
+$lang['Gammu version'] = 'Gammu Version';
+$lang['Gammu DB schema'] = 'Gammu DB Schema';
+$lang['Modem IMEI'] = 'Modem IMEI';
 
 // Phonebook
-$lang['kalkun_contact'] = 'Kontakte';
-$lang['kalkun_all_contacts'] = 'All Contacts';
-$lang['kalkun_send_to_all_contacts'] = 'Send To All Contacts';
-$lang['kalkun_group'] = 'Gruppen';
-$lang['kalkun_add_to_group'] = 'Zu Gruppe hinzufügen';
-$lang['kalkun_delete_from_group'] = 'Von Gruppe löschen';
-$lang['kalkun_pbk_add_method'] = 'Methode zum hinzufügen von Kontakten wählen';
-$lang['kalkun_pbk_add_form'] = 'Kontaktformular benutzen';
-$lang['kalkun_pbk_add_form_desc'] = 'Kontakt händische mithilfe des Kontaktformulars hinzufügen';
-$lang['kalkun_pbk_add_csv'] = 'CSV-Datei importieren';
-$lang['kalkun_pbk_add_csv_desc'] = 'Kontakt aus CSV-Datei importieren';
-$lang['kalkun_pbk_add_csv_file'] = 'CSV Datei';
-$lang['kalkun_pbk_add_csv_label'] = 'Die CSV-Datei muss in einem gültigen Format vorliegen';
-$lang['kalkun_pbk_del_contact_conf'] = "Sind Sie sicher, dass sie <span id='contact-delete-count'></span> Kontakt(e) löschen wollen?";
-$lang['kalkun_valid_example'] = 'Beispiel';
+$lang['Contact'] = 'Kontakte';
+$lang['All contacts'] = 'All Contacts';
+$lang['Send to all contacts'] = 'Send To All Contacts';
+$lang['Add to group'] = 'Zu Gruppe hinzufügen';
+$lang['Delete from group'] = 'Von Gruppe löschen';
+$lang['Select add contact method'] = 'Methode zum hinzufügen von Kontakten wählen';
+$lang['Manual input'] = 'Kontaktformular benutzen';
+$lang['Manually add contact using contact form'] = 'Kontakt händische mithilfe des Kontaktformulars hinzufügen';
+$lang['From CSV file'] = 'CSV-Datei importieren';
+$lang['Import contact from CSV format file'] = 'Kontakt aus CSV-Datei importieren';
+$lang['CSV file'] = 'CSV Datei';
+$lang['The CSV file must be in valid format'] = 'Die CSV-Datei muss in einem gültigen Format vorliegen';
+$lang['Are you sure want to delete <span id=\'contact-delete-count\'></span> contact(s)?'] = "Sind Sie sicher, dass sie <span id='contact-delete-count'></span> Kontakt(e) löschen wollen?";
```

Comment:
Does this look too long? Specifically html inside the keys. IMO It would be better with the constants like `kalkun_pbk_del_contact_conf` which is simpler to understand and still allows translations to happen/be managed.  Another problem is that if that string is ever changed, then both the key and value would need to updated for all languages, vs only updating particular language or even just the values.

A very good example of this is how android apps have multiple languages and use constant_keys for accessing them. https://developer.android.com/training/basics/supporting-devices/languages

Reply:
- I checked and there is no problem with key length in php, there is almost no limit or it is extremely high IIRC.
- Actually, if the english text changes, the translation has to be changed too, so this is not a point in IMO.
- The new way I propose permits to reuse existing translation more easily since if the word/label already exists, it will be used automatically and there is no need to search if there is already such a label. BTW reusing a label as you point (let's say Delete = kalkun_pbk_del_contact_conf, if I want to reuse delete in a delete button for a user, putting the key kalkun_pbk_del_contact_conf would make the understanding worse, or there would be duplicate translations are there already were in the code)

Reply:
Ah, I see, these were based on how Qt works. 

I was trying to find a few i18n libraries that are available in PHP and found this one https://github.com/Philipp15b/php-i18n, and this looks this is used by a lot of projects. So based on this IMO it makes sense to have our implementation in such a way so that we can migrate to non-PHP files for translation in future.

We don't need to say `kalkun_pbk_del_contact_conf = Delete` but a simple. `delete = Delete` allows it to be used in every place, and in the same way `add_contact_using_form=Manually add contact using contact form` makes it readable, reusable.

 
Also the examples from https://codeigniter.com/user_guide/outgoing/localization.html talks on the same lines.

Reply:
Sind the change took time to be fine especially wrt the duplicates it is unlikely I modify it.
Btw, I'm not fond of adding libraries for everything. However, on another subject, I think there is one that should be used, albeit it is large: libphonenumber.

Reply:
Any chance to see that merged as it is?

Reply:
I have a few proposals in this regard, need some time to write them up. I hope that is okay.